workflows/rnaseq-ref-index/make_pre_mrna_fasta.R

```diff
@@ -123,14 +140,46 @@ eisaR::exportToGtf(
   filepath = spliced_intron_gtf
 )
 
-# create text file mapping transcript and intron identifiers to 
-# corresponding gene identifiers
-# first make Tx2Gene for all spliced and intron sequences
+# create text file mapping transcript and intron identifiers to corresponding gene identifiers
+# make 2 column Tx2Gene for all spliced and intron sequences
 full_tx2gene <- eisaR::getTx2Gene(
   grl, filepath = spliced_intron_tx2gene
 )
 
-## need to write out to metadata 
+# trim grl used for splici
+splici_grl <- trim(grl)
+
+# extract seqs from trimmed grl for splici index
+splici_seqs <- GenomicFeatures::extractTranscriptSeqs(
+  x = genome, 
+  transcripts = splici_grl
+)
+
+# remove sequence duplicates
+splici_seqs <- unique(splici_seqs)
+splici_grl <- splici_grl[names(splici_seqs)]
+
+# write splici to fasta 
+Biostrings::writeXStringSet(
+  splici_seqs, filepath = splici_fasta, compress = TRUE
+)
+
+# make 3 column Tx2 gene needed for alevin-fry USA mode
+splici_tx2gene_df <- eisaR::getTx2Gene(splici_grl)
+
+splici_tx2gene_df[, "status"] = sapply(
+  strsplit(splici_tx2gene_df$transcript_id, "-"), function(x) 
+    if(length(x) == 2){"U"} else {"S"}
+)
+
+splici_tx2gene_df[, "gene_id"] = sapply(
+  strsplit(splici_tx2gene_df$gene_id, "-"), function(x) x[1]
+)
```

Comment:
I think I would rather see these as tidyverse commands for clarity. 

For example, I am trying to understand teh status column here, and I think the idea is that if there is a `-` in the transcript id that indicates it is unspliced, but that is *so* not how I would write it...

```suggestion
splici_tx2gene_df <- splici_tx2gene_df  %>%
  mutate(status = ifelse(str_count(transcript_id, '-') > 0, 'U', 'S')
         gene_id = str_extract(gene_id, "[^-]+")
```

Check that this does the same thing, of course, but I think it does?

Reply:
I would also prefer this ;).  Would you be willing to contribute your modifications back to the function we have for this in our repo (https://github.com/COMBINE-lab/usefulaf) ?

Reply:
Sure! I can't wait to have that repo name on my list! 🤣

Reply:
When updating, I remembered another function that I always forget about:`stringr::word()`. 

So I would even more prefer
```
gene_id = word(gene_id, 1, sep = '-')
```
to the `str_extract` version above.